### ISSUE TITLE
Unable To Launch Games - Kinetic OR Lunar - Steam Snap Edge

### ISSUE THREAD

**Issue by storyaddict** [canonical/steam-snap#126]
With multiple Steam games installed locally, clicking play will act as though the game is about to launch, then interface returns to the play button.

NOTE: this does **NOT** happen with .deb version. This issue is present on both a fresh install of Kinetic Kudu (22.10), and after upgrading to Lunar Lobster (23.04).

I tested with Cyberpunk 2077, Mass Effect Legendary Edition, and Sonic Frontiers. All games were stored on a 2nd hard drive formatted as ext4.

Proton settings set to Experimental.

**To Reproduce**

1. Open Steam.
2. Click on any locally installed game.
3. Click "Play" - note vulkan shaders window may pop up, or EULA page, or both.
4. Game fails to launch, "Stop" button changes back to "Play" automatically.

**Expected behavior**

1. Click Play.
2. Agree to EULA, wait for vulkan shaders, etc. 
3. Game launches.

**Environment**

`snap run steam.report`

```
os_release: 
    name:               "Ubuntu"
    version:            "23.04 (Lunar Lobster)"
snap_info: 
    steam_revision:     124
    snapd_revision:     18357
lspci: 
    01:00.0:            NVIDIA Corporation GP104BM [GeForce GTX 1070 Mobile] (rev a1)
glxinfo: 
    gpu:                NVIDIA GeForce GTX 1070/PCIe/SSE2
    gpu_version:        4.6.0 NVIDIA 525.85.05
lscpu: 
    model_name:         Intel(R) Core(TM) i7-7820HK CPU @ 2.90GHz
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu
```

**Logs**

I suspect the issue is tied to THIS output:

```
Mar 09 16:14:46 nathaniel-Oryx-Pro-3 steam_steam.desktop[11498]: bwrap: Can't bind mount /oldroot/media/nathaniel/Extra Drive 1/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier/var/tmp-F71C11/usr on /newroot/usr: Unable to mount source on destination: Permission denied
Mar 09 16:14:46 nathaniel-Oryx-Pro-3 steam_steam.desktop[9006]: GameAction [AppID 1328670, ActionID 9] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
```

Full capture of relevant journalctl follows.

**Comment by storyaddict** [comment]
Logs Attached:
[kinetic-steam-test-logs.txt](https://github.com/canonical/steam-snap/files/10949371/kinetic-steam-test-logs.txt)
[lunar-steam-test-logs.txt](https://github.com/canonical/steam-snap/files/10949372/lunar-steam-test-logs.txt)

**Comment by kenvandine** [comment]
Thanks for the report.  Please provide the output of "snap connections steam"

**Comment by storyaddict** [comment]
Sure, here it is:

```
nathaniel@nathaniel-Oryx-Pro-3:~$ snap connections steam
Interface               Plug                          Slot                                Notes
audio-playback          steam:audio-playback          :audio-playback                     -
audio-record            steam:audio-record            -                                   -
bluez                   steam:bluez                   -                                   -
content[gaming-mesa]    steam:gaming-mesa             gaming-graphics-core22:gaming-mesa  -
content[gtk-3-themes]   steam:gtk-3-themes            gtk-common-themes:gtk-3-themes      -
content[icon-themes]    steam:icon-themes             gtk-common-themes:icon-themes       -
content[sound-themes]   steam:sound-themes            gtk-common-themes:sound-themes      -
desktop                 steam:desktop                 :desktop                            -
desktop-legacy          steam:desktop-legacy          :desktop-legacy                     -
fuse-support            steam:fuse-support            -                                   -
gsettings               steam:gsettings               :gsettings                          -
hardware-observe        steam:hardware-observe        :hardware-observe                   manual
home                    steam:home                    :home                               -
joystick                steam:joystick                -                                   -
mount-observe           steam:mount-observe           -                                   -
network                 steam:network                 :network                            -
network-control         steam:network-control         -                                   -
opengl                  steam:opengl                  :opengl                             -
personal-files          steam:dot-local-share-steam   -                                   -
process-control         steam:process-control         -                                   -
removable-media         steam:removable-media         :removable-media                    -
screen-inhibit-control  steam:screen-inhibit-control  :screen-inhibit-control             -
shared-memory           steam:shmem                   :shared-memory                      -
steam-support           steam:steam-support           :steam-support                      -
system-observe          steam:system-observe          :system-observe                     manual
upower-observe          steam:upower-observe          :upower-observe                     -
wayland                 steam:wayland                 :wayland                            -
x11                     steam:x11                     :x11        
```

**Comment by Nightmayr** [comment]
Seeing the same error. Preventing me from playing Proton games
```
bwrap: Can't bind mount /oldroot/media/umayr/Games/steam/steamapps/common/SteamLinuxRuntime_soldier/var/tmp-GOOS11/usr on /newroot/usr: Unable to mount source on destination: Permission denied
Game process removed: AppID 1172470 "/home/umayr/snap/steam/common/.local/share/Steam/ubuntu12_32/reaper SteamLaunch AppId=1172470 -- /home/umayr/snap/steam/common/.local/share/Steam/ubuntu12_32/steam-launch-wrapper -- '/media/umayr/Games/steam/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/media/umayr/Games/steam/steamapps/common/Proton - Experimental'/proton waitforexitandrun  '/media/umayr/Games/steam/steamapps/common/Apex Legends/EasyAntiCheat_launcher.exe' -steam", ProcID 7230 
```

**Native** games like Half Life 2 work for me.

Steam version info:

```
installed:          1.0.0.75            (124) 188MB -
```

Snapd version info

```
installed:          2.58.2                          (18357) 52MB snapd
```

System specs:
```
GPU: NVIDIA GeForce RTX 2080 Ti/PCIe/SSE2
GPU Driver:  4.6.0 NVIDIA 525.85.05
OS: Ubuntu 22.10
DE: GNOME
```

Steam connections:

```
$ snap connections steam
Interface               Plug                          Slot                                Notes
audio-playback          steam:audio-playback          :audio-playback                     -
audio-record            steam:audio-record            -                                   -
bluez                   steam:bluez                   :bluez                              manual
content[gaming-mesa]    steam:gaming-mesa             gaming-graphics-core22:gaming-mesa  -
content[gtk-3-themes]   steam:gtk-3-themes            gtk-common-themes:gtk-3-themes      -
content[icon-themes]    steam:icon-themes             gtk-common-themes:icon-themes       -
content[sound-themes]   steam:sound-themes            gtk-common-themes:sound-themes      -
desktop                 steam:desktop                 :desktop                            -
desktop-legacy          steam:desktop-legacy          :desktop-legacy                     -
fuse-support            steam:fuse-support            -                                   -
gsettings               steam:gsettings               :gsettings                          -
hardware-observe        steam:hardware-observe        -                                   -
home                    steam:home                    :home                               -
joystick                steam:joystick                :joystick                           manual
mount-observe           steam:mount-observe           -                                   -
network                 steam:network                 :network                            -
network-control         steam:network-control         -                                   -
opengl                  steam:opengl                  :opengl                             -
personal-files          steam:dot-local-share-steam   -                                   -
process-control         steam:process-control         -                                   -
removable-media         steam:removable-media         :removable-media                    -
screen-inhibit-control  steam:screen-inhibit-control  :screen-inhibit-control             -
shared-memory           steam:shmem                   :shared-memory                      -
steam-support           steam:steam-support           :steam-support                      -
system-observe          steam:system-observe          -                                   -
upower-observe          steam:upower-observe          :upower-observe                     -
wayland                 steam:wayland                 :wayland                            -
x11                     steam:x11                     :x11                                -
```

**Comment by blubberdiblub** [comment]
Can confirm. It's only Proton games that don't work (tried switching between Proton 6.3, 7.0 and Experimental without success). Natively running games work fine. Proton games worked fine with the `steam-launcher` package I had installed before replacing it with the snap.

**Comment by lunarcloud** [comment]
Confirm. 
Can't run majority of my games, as they run via Proton. 
Was forced to switch to the snap since updating to 23.04 broke the deb version...

**Comment by ca1ko** [comment]
I got this exact problem after a fresh install today but it was on 22.04, anything that relied on proton worked after importing my library.

However, I was able to run my library without any issues when I reinstalled the snap package but before importing my library, I enabled proton and installed a game in the default location inside the snap folder.

To my surprise, it did run without any issues. So, I then imported my library to give it a try, and everything began to work again. Hope this helps!

**Comment by Nightmayr** [comment]
> However, I was able to run my library without any issues when I reinstalled the snap package but before importing my library, I enabled proton and installed a game in the default location inside the snap folder.

Thanks for posting your findings, this worked for me as well. Similarly if you:

1.  Move a game that uses proton from the external drive library over to the default location used by the snap (I opted for something small like Among Us)
2.  Launch the game that was moved
3.  Quit the Game
4.  Launch a game stored on the external library

I've found that this also allows proton games to launch. Hopefully this is useful information/can provide clues on how to solve this problem so that external drive libraries can be added and played without needing to do any of these steps like the deb.
